### PR TITLE
[NEW] WebSocket connection status on the title view

### LINKED ISSUE #1740: [BUG] Subscription list title view is left aligned on iOS 10
When running the app with the new design on iOS 10 simulators the title view of the subscription list is left aligned, as we can see on the screen shot below:

screen shot 2018-06-01 at 15 50 29

### LINKED ISSUE #1752: [NEW] Add "Connecting..." or "Waiting for network..." states on title views

### PR DESCRIPTION
**Pull request by rafaelks**
@RocketChat/ios

Closes #1740
Closes #1752 

## Progress

- [x] Subscriptions list
- [x] Chat controller
- [x] iPad
- [x] iOS 10
- [x] i18n
- [x] Tests

## Preview

[Click here to see the preview of this PR](https://drive.google.com/file/d/1JlcvPHwtnM8Tuv2cKlXbhMN6fuQILCDE/view?usp=sharing)

### CODE REVIEW

File: Rocket.Chat/Controllers/Subscriptions/SubscriptionsViewController.swift [hunk 1]
```diff
@@ -272,14 +274,18 @@ final class SubscriptionsViewController: BaseViewController {
 
     func setupTitleView() {
         if let titleView = SubscriptionsTitleView.instantiateFromNib() {
+            titleView.translatesAutoresizingMaskIntoConstraints = false
+            titleView.delegate = self
+            titleView.layoutIfNeeded()
+            titleView.sizeToFit()
+            updateServerInformation()
+
+            titleView.translatesAutoresizingMaskIntoConstraints = true
```
Comment (filipealva) [inline Rocket.Chat/Controllers/Subscriptions/SubscriptionsViewController.swift]: @rafaelks why exactly are you setting it back to `true`?

Comment (rafaelks) [reply Rocket.Chat/Controllers/Subscriptions/SubscriptionsViewController.swift]: iOS 10 hacks 🤓

Comment (rafaelks) [reply Rocket.Chat/Controllers/Subscriptions/SubscriptionsViewController.swift]: Reference: https://stackoverflow.com/questions/15285698/building-a-titleview-programmatically-with-constraints-or-generally-constructin

Comment (filipealva) [reply Rocket.Chat/Controllers/Subscriptions/SubscriptionsViewController.swift]: Interesting 🤔 maybe it worth it to put it as a comment above this line so everyone can easily understand and to make sure we will remember to remove it when we stop supporting iOS 10

**Comment by filipealva** [comment]
The changes are looking good, but I found some issues when testing it on my device:

- When I try to login I get stuck on login screen loading forever. I look at the logs and it looks like I’m already authenticated, but nothing happen. I have to close the app and open it again, then it opens already authenticated.


- Sometimes when I put the app in background and bring it back to foreground the title view gets stuck on “connecting” state. The logs show a generic socket error saying that it got disconnected. I had to close the app to make it connect again. Weird. Happened 2 times while testing, but doesn’t happen always. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**